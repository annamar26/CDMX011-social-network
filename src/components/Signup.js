/* eslint-disable no-undef */
import { fbFunctions } from '../firebaseClient.js';
import { onNavigate } from '../routes.js';

const signup = () => {
  const container = document.createElement('div');
  const html = `<nav><div>
    <img class='logo logo1' src='./icons&img/logo.png'></div>
   
    </nav>
<div class='signUp-container'>
    
    <h2>Regístrate para empezar</h2>
    <form id='registerForm'>
      <label for='sgEmail'>Email</label>
    <input id='sgEmail' type=email required>
    <label for='sgPassword'>Contraseña</label>
    <input id='sgPassword' type='password' required><img class='s-button' src='./icons&img/ver.svg'>
    <p class='warning' id='errorMessage'></p>
    <input id='signUpSubmit' type=submit value='Crear cuenta'>
      </form>
      <img class='div' src='./icons&img/div.svg'>
      <button id='googleS' class='button'><img class='gicon' src='./icons&img/googleicon.svg'>Regístrate con Google</button>
<img id='return' class='return'src='./icons&img/return.svg'>
</div>`;

  container.innerHTML = html;

  container.querySelector('#googleS').addEventListener('click', (e) => {
    e.preventDefault();
    fbFunctions.googleUserSignUp(e)
      .then(() => {
        onNavigate('/timeline');
        fbFunctions.getPosts();
        fbFunctions.createUserDoc();
      })
      .catch((error) => { container.querySelector('#errorMessage').innerHTML = error; });
  });

  container.querySelector('#return').addEventListener('click', (e) => {
    e.preventDefault();
    onNavigate('/');
  });

  container.querySelector('.s-button').addEventListener('click', (e) => {
    e.preventDefault();

    const tipo = container.querySelector('#sgPassword');
    if (tipo.type === 'password') {
      tipo.type = 'text';
    } else {
      tipo.type = 'password';
    }
  });

  const signUpData = container.querySelector('#registerForm');
  signUpData.addEventListener('submit', (e) => {
    e.preventDefault();
    const signupEmail = container.querySelector('#sgEmail').value;
    const signupPassword = container.querySelector('#sgPassword').value;
    fbFunctions.userSignup(signupEmail, signupPassword).then(() => {
      onNavigate('/profile');
      fbFunctions.getCurrentUserPosts();
      fbFunctions.createUserDoc();
    })
      .catch((error) => { container.querySelector('#errorMessage').innerHTML = error; });
  });
  return container;
};
export default signup;
