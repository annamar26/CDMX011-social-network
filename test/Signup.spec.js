/**
 * @jest-environment jsdom
 */
import Signup from '../src/components/Signup.js';
import { render } from '../src/utils.js';

describe('Signup', () => {
  document.body.innerHTML = '<div id="root"></div>';
  const rootDiv = document.getElementById('root');
  it('should render Signup page', () => {
    render(rootDiv, Signup());
    expect(rootDiv).toMatchSnapshot();
  });
});
