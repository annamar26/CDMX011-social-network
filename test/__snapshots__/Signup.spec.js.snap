// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should render Signup page 1`] = `
"() =&gt; {
  /* istanbul ignore next */
  cov_17uxal9kqk().f[0]++;
  const container =
  /* istanbul ignore next */
  (cov_17uxal9kqk().s[1]++, document.createElement('div'));
  const html =
  /* istanbul ignore next */
  (cov_17uxal9kqk().s[2]++, \`&lt;nav&gt;
    &lt;img class='logo' src='./icons&amp;img/logo.png'&gt;
   
    &lt;/nav&gt;
&lt;div class='signUp-container'&gt;
    
    &lt;h2&gt;Regístrate para empezar&lt;/h2&gt;
    &lt;form id='registerForm'&gt;
    &lt;label for='username'&gt;Nombre de usuario&lt;/label&gt;
    &lt;input id='username' type=text  required&gt;
    &lt;label for='sgEmail'&gt;Email&lt;/label&gt;
    &lt;input id='sgEmail' type=email required&gt;
    &lt;label for='sgPassword'&gt;Contraseña&lt;/label&gt;
    &lt;input id='sgPassword' type='password' required&gt;&lt;img id='seeBn'class='s-button' src='./icons&amp;img/ver.svg'&gt;
    &lt;p class='warning' id='errorMessage'&gt;&lt;/p&gt;
    &lt;input id='signUpSubmit' type=submit value='Crear cuenta'&gt;
      &lt;/form&gt;
      &lt;img class='div' src='./icons&amp;img/div.svg'&gt;
      &lt;button id='googleS' class='button'&gt;&lt;img class='gicon' src='./icons&amp;img/googleicon.svg'&gt;Regístrate con Google&lt;/button&gt;
&lt;img id='return' class='return'src='./icons&amp;img/return.svg'&gt;
&lt;/div&gt;\`);

  /* istanbul ignore next */
  cov_17uxal9kqk().s[3]++;
  container.innerHTML = html;

  /* istanbul ignore next */
  cov_17uxal9kqk().s[4]++;
  container.getElementById('googleS').addEventListener('click', (e) =&gt; {
    /* istanbul ignore next */
    cov_17uxal9kqk().f[1]++;
    cov_17uxal9kqk().s[5]++;
    e.preventDefault();

    /* istanbul ignore next */
    cov_17uxal9kqk().s[6]++;

    /* istanbul ignore next */
    _firebaseClient.
    /* istanbul ignore next */
    fbFunctions.googleUserSignUp(e).then(() =&gt; {
      /* istanbul ignore next */
      cov_17uxal9kqk().f[2]++;
      cov_17uxal9kqk().s[7]++;
      console.log('Google singin');

      /* istanbul ignore next */
      cov_17uxal9kqk().s[8]++;

      /* istanbul ignore next */
      (0,
      /* istanbul ignore next */
      _routes.
      /* istanbul ignore next */
      onNavigate)('/timeline');

      /* istanbul ignore next */
      cov_17uxal9kqk().s[9]++;

      /* istanbul ignore next */
      _firebaseClient.
      /* istanbul ignore next */
      fbFunctions.getPosts();
    }).catch((error) =&gt; {
      /* istanbul ignore next */
      cov_17uxal9kqk().f[3]++;
      cov_17uxal9kqk().s[10]++;
      container.querySelector('#errorMessage').innerHTML = error;
    });
  });

  /* istanbul ignore next */
  cov_17uxal9kqk().s[11]++;
  container.getElementById('return').addEventListener('click', (e) =&gt; {
    /* istanbul ignore next */
    cov_17uxal9kqk().f[4]++;
    cov_17uxal9kqk().s[12]++;
    e.preventDefault();

    /* istanbul ignore next */
    cov_17uxal9kqk().s[13]++;

    /* istanbul ignore next */
    (0,
    /* istanbul ignore next */
    _routes.
    /* istanbul ignore next */
    onNavigate)('/');
  });

  /* istanbul ignore next */
  cov_17uxal9kqk().s[14]++;
  container.querySelector('.s-button').addEventListener('click', (e) =&gt; {
    /* istanbul ignore next */
    cov_17uxal9kqk().f[5]++;
    cov_17uxal9kqk().s[15]++;
    e.preventDefault();
    const tipo =
    /* istanbul ignore next */
    (cov_17uxal9kqk().s[16]++, container.getElementById('sgPassword'));

    /* istanbul ignore next */
    cov_17uxal9kqk().s[17]++;

    if (tipo.type === 'password') {
      /* istanbul ignore next */
      cov_17uxal9kqk().b[0][0]++;
      cov_17uxal9kqk().s[18]++;
      tipo.type = 'text';
    } else {
      /* istanbul ignore next */
      cov_17uxal9kqk().b[0][1]++;
      cov_17uxal9kqk().s[19]++;
      tipo.type = 'password';
    }
  });
  const signUpData =
  /* istanbul ignore next */
  (cov_17uxal9kqk().s[20]++, container.querySelector('#registerForm'));

  /* istanbul ignore next */
  cov_17uxal9kqk().s[21]++;
  signUpData.addEventListener('submit', (e) =&gt; {
    /* istanbul ignore next */
    cov_17uxal9kqk().f[6]++;
    cov_17uxal9kqk().s[22]++;
    e.preventDefault();
    const signupEmail =
    /* istanbul ignore next */
    (cov_17uxal9kqk().s[23]++, document.querySelector('#sgEmail').value);
    const signupPassword =
    /* istanbul ignore next */
    (cov_17uxal9kqk().s[24]++, document.querySelector('#sgPassword').value);

    /* istanbul ignore next */
    cov_17uxal9kqk().s[25]++;

    /* istanbul ignore next */
    _firebaseClient.
    /* istanbul ignore next */
    fbFunctions.userSignup(signupEmail, signupPassword).then((user) =&gt; {
      /* istanbul ignore next */
      cov_17uxal9kqk().f[7]++;
      cov_17uxal9kqk().s[26]++;
      console.log(user);

      /* istanbul ignore next */
      cov_17uxal9kqk().s[27]++;
      console.log('Login exitoso');

      /* istanbul ignore next */
      cov_17uxal9kqk().s[28]++;

      /* istanbul ignore next */
      (0,
      /* istanbul ignore next */
      _routes.
      /* istanbul ignore next */
      onNavigate)('/profile');

      /* istanbul ignore next */
      cov_17uxal9kqk().s[29]++;

      /* istanbul ignore next */
      _firebaseClient.
      /* istanbul ignore next */
      fbFunctions.getCurrentUserPosts();
    }).catch((error) =&gt; {
      /* istanbul ignore next */
      cov_17uxal9kqk().f[8]++;
      cov_17uxal9kqk().s[30]++;
      document.getElementById('errorMessage').innerHTML = error;
    }); // this.userName = document.querySelector('#username').value;
    // fbFunctions.addUserDisplayName(this.userName);

  });

  /* istanbul ignore next */
  cov_17uxal9kqk().s[31]++;
  return container;
}"
`;
